### PR TITLE
Feature/commit issue action

### PR DESCRIPTION
**Pull request by wescran**
### Overview
This feature introduces a workflow using Github Actions. The purpose of this workflow is to look for recent commits on the master branch to [`tree-sitter/tree-sitter-python`](https://github.com/tree-sitter/tree-sitter-python) and create an issue in this repository if one is found.

### Notes
The workflow `commit-issue.yml` is set to run every hour and the `entrypoint.sh` script that the action uses checks the previous hour for commits using the GitHub API. These settings can be tweaked by changing the cron syntax in the workflow and by adjusting the date syntax in the entrypoint script.

```
name: Commit Issue Workflow
on:
  schedule:
    - cron: '0 * * * *' #runs every hour
```
```
since=$(date -u -d '1 hour ago' +"%Y-%m-%dT%H:%M:%SZ") #only commits after this date returned
until=$(date -u +"%Y-%m-%dT%H:%M:%SZ") #only commits before this date returned 
```
Information on the inputs for the action can be found in the `README.md`

#### Future work
I will probably add more input variables to allow for easier adjustments of the date variables 

Solves #11

### CODE REVIEW

File: .github/actions/commit-issue-action/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# commit-issue-action
+**Based off of [commit-issue-action](https://github.com/wescran/commit-issue-action)**
+
+GitHub action that monitors public repositories for commits to the master branch and creates an issue if found. There are 3 inputs that can be written to the workflow `commit-issue.yml` found in the `.github` directory.
+
+## Inputs
+
+### `sourceRepo`
+
+**Required** The name of the repository to monitor for commits. Should be in the form of `{owner}/{repo}`
```
Comment (madiedgar) [inline .github/actions/commit-issue-action/README.md:10]: Does this monitor for any activity or just commits on the master branch?

**Review by madiedgar — COMMENTED** [review]
@wescran this looks amazing! Great work!! I'm not as familiar with shell scripts, so forgive me if this is redundant. 

>The script checks the sourceRepo for any recent commits within the past hour. If found, then it will create an issue in the targetRepo. The script runs every hour only checking for recent commits within the past hour, so issue duplicates wouldn't occur. 

Is that correct?

File: .github/workflows/commit-issue.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      - uses: actions/checkout@v1
+      - name: Commit issue action step
+        id: issue
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        uses: ./.github/actions/commit-issue-action
+        with:
+          sourceRepo: 'tree-sitter/tree-sitter-python'
+          targetRepo: 'legesher/tree-sitter-legesher-python'
+          issueLabels: 'Type: Maintenance, Status: Available, Status: Review Needed'
```
Comment (madiedgar) [inline .github/workflows/commit-issue.yml]: Would you mind changing the issue label `Type: Maintenance` to be `🚧 Opportunity: Maintenance`? We recently updated our labels.

**Review by madiedgar — CHANGES_REQUESTED** [review]
Also, due to the language change, numpy file examples throw errors. You can properly test if this breaks anything by adding `examples/numpy/benchmarks/benchmarks/bench_avx.py` to the `./script/known_failures.txt` file.

File: .github/actions/commit-issue-action/README.md (hunk 1 above)
Comment (wescran) [reply .github/actions/commit-issue-action/README.md:10]: Right now it just monitors for commits to the master branch.

File: .github/workflows/commit-issue.yml (hunk 2 above)
Comment (wescran) [reply .github/workflows/commit-issue.yml]: Yup, I can do that!

**Comment by wescran** [comment]
> @wescran this looks amazing! Great work!! I'm not as familiar with shell scripts, so forgive me if this is redundant.
> 
> > The script checks the sourceRepo for any recent commits within the past hour. If found, then it will create an issue in the targetRepo. The script runs every hour only checking for recent commits within the past hour, so issue duplicates wouldn't occur.
> 
> Is that correct?

Yes that is a correct way of describing the script!


> Also, due to the language change, numpy file examples throw errors. You can properly test if this breaks anything by adding `examples/numpy/benchmarks/benchmarks/bench_avx.py` to the `./script/known_failures.txt` file.

Yup I can do that as well!

**Comment by wescran** [comment]
> > @wescran this looks amazing! Great work!! I'm not as familiar with shell scripts, so forgive me if this is redundant.
> > > The script checks the sourceRepo for any recent commits within the past hour. If found, then it will create an issue in the targetRepo. The script runs every hour only checking for recent commits within the past hour, so issue duplicates wouldn't occur.
> > 
> > 
> > Is that correct?
> 
> Yeah that is redundant, I can revise that.

Woops, sorry! Ignore what I said, I thought it was something I had written. Yes that is a correct way of describing the script.

File: .github/actions/commit-issue-action/README.md (hunk 1 above)
Comment (madiedgar) [reply .github/actions/commit-issue-action/README.md:10]: Perfect!!

**Review by madiedgar — COMMENTED** [review]
Beautiful!!! 👌

**Pull request merged**